### DISCUSSION TITLE
Text-Index

### DISCUSSION THREAD

**Discussion by filipecaramelo** [GMOD/jbrowse-components#3055] [Q&A]
Hello everyone, I'm using JBrowse2 to create a little interactive hub for our group at the lab, but there is something can't I can't really turn my head around on.

I made several genomes with annotations already using text-index and they work, somehow in some other genomes when I used the exact same command I can't seem to make this search to work.

I noticed that whenever I search something at the bar it returns a : Error: ${this.url} fetch returned status 200, expected 206 — react-linear-genome-view.umd.development.js:152106

I made sure every path was correct and I even have the Dist folder inside my root page folder, is there something I can do to make this work? 

If there is any question I can answer to make It easier for you guys to understand my problem let me know, im really tryin to think of every possibility that could be leading to this tho.

**Comment by cmdcolin** [comment]
Hi there, good question. Just as background on what the error means: it is a fairly strict check that tries to ensure, if we do a range request, that the web server returns a range response (HTTP status 206) back. If the web server returns HTTP status 200, it could indicate that the range request was not properly understood by the web server. What type of web server are you using to host your jbrowse 2 instance?

**Reply by cmdcolin** [reply]
If it is responding HTTP status 206 to other requests properly and it is just this one, it could be your server is returning a HTTP status 200 for a 404...could check in the network logs to see if this is the case (e.g. look at network log in chrome devtools, see the request for the ix file, copy the url it is using and navigate to it in your web browser and see if it is actually the ix file)

**Reply by filipecaramelo** [reply]
> Hi there, good question. Just as background on what the error means: it is a fairly strict check that tries to ensure, if we do a range request, that the web server returns a range response (HTTP status 206) back. If the web server returns HTTP status 200, it could indicate that the range request was not properly understood by the web server. What type of web server are you using to host your jbrowse 2 instance?

Hello, to my completely honest, I don't know, all I know is that it is running an Apache server, I don't know the version or anything.

I tried the Chrome dev tools, the link of the file is correct, for all of the files

RefNameAutocomplete.tsx:135 Error: ${this.url} fetch returned status 200, expected 206
    at RemoteFileWithRangeCache.read (remoteFile.js:110:1)
    at async Trix._getBuffer (index.js:113:1)
    at async Trix.search (index.js:24:1)

this was what came up in the console, something that I noticed was that genomes made with an older version of the cli (before I updated it) don't seem to be making a problem while new ones that I added are the ones making the issue, could this be related to the cli version I used?

**Reply by filipecaramelo** [reply]
I even checked the .ix, .ixx and meta files thinking that could be a mistake on the output of the files, but no they have the same structure and everything.


I even used the same exact Js + html structure in both visualizations, while the scaffold search works the indexed does not

**Reply by cmdcolin** [reply]
hmmm. don't know if i have too many clues! it looks like you are using embedded. it is a bit trickier to configure embedded since it is generally done by hand, but this example on our storybook docs is probably our best reference for it https://jbrowse.org/storybook/lgv/main/?path=/story/text-searching--page

that would be quite weird that an older version would work while the newer one would have issue. if you know the exact working version, broken version, could try to look more into that perhaps

**Reply by filipecaramelo** [reply]
Ok just retried it with @jbrowse/cli/1.4.4 darwin-x64 node-v14.17.6 , with the exact same steps and works amazingly, I was using the latest version before

**Reply by cmdcolin** [reply]
is the jbrowse version that you are are using also up to date? i saw it looked like you were using embedded, it is tricky to check but i would check this.

if it is not the latest version, the following issue could occur: the latest version of jbrowse increased the "prefix size" on trix indexes somewhat, and the older versions of jbrowse would not be able to understand this updated prefix size to be larger (https://jbrowse.org/jb2/blog/2022/04/26/v1.7.5-release/) and so the CLI by default outputted trix (ixx) files prefix size 6 instead of 5, but then older versions of the jbrowse client would not understand these newer ixx files (they have a hardcoded limit on prefix size 5) and this hardcoded prefix would actually cause misinterpretation of the file and could lead to this issue

**Reply by filipecaramelo** [reply]
ahhhh, man, might be that!! im not sure if I updated everything and that would make total sense to me now. it is a bit tricky indeed but I do love your software so I don't mind get around if it means me being able to show the lab some cool genomes hehe :)

I'll check everything up and make sure I update everything within the app, thanks for the support Colin :) 👍
